#==================================================================================================
# Patch to add formal Quartz build option; required to easily toggle between Quartz and X11
#==================================================================================================
--- meson_options.txt.orig	2026-09-12 10:57:41.000000000 -0400
+++ meson_options.txt	2026-09-12 10:59:09.000000000 -0400
@@ -1,4 +1,5 @@
 # Build properties
+option('quartz',            type: 'boolean', value: true,   description: 'Enable quartz for macOS')
 option('ansi',              type: 'boolean', value: false,  description: 'Turn on strict ansi')
 option('enable-console-bin',type: 'boolean', value: true,   description: 'Build a console-only binary which does not link GTK')
 option('enable-multiproc',  type: 'boolean', value: true,   description: 'Support for multiple processors')
--- meson.build.orig	2026-09-12 10:57:34.000000000 -0400
+++ meson.build	2026-09-12 11:00:58.000000000 -0400
@@ -497,7 +497,8 @@
 
 gtk3_minver       = '3.24.0'
 gtk3              = dependency('gtk+-3.0',           version: '>='+gtk3_minver)
-gdk_specific_name = platform_osx ? 'gdk-quartz-3.0' : 'gdk-3.0'
+gdk_quartz        = get_option('quartz')
+gdk_specific_name = gdk_quartz ? 'gdk-quartz-3.0' : 'gdk-3.0'
 gdk_specific      = dependency(gdk_specific_name)
 gtk_version_split = gtk3_minver.split('.')
 gtk_major = gtk_version_split[0].to_int()
